diff --git a/deps/c-ares/configure b/deps/c-ares/configure
index ca35a191..50c8266b 100755
--- a/deps/c-ares/configure
+++ b/deps/c-ares/configure
@@ -31859,7 +31859,7 @@ fi



-ac_config_files="$ac_config_files Makefile include/Makefile src/Makefile src/lib/Makefile src/tools/Makefile docs/Makefile libcares.pc"
+ac_config_files="$ac_config_files Makefile include/Makefile src/Makefile src/lib/Makefile src/tools/Makefile libcares.pc"


 cat >confcache <<\_ACEOF
@@ -33008,7 +33008,6 @@ do
     "src/Makefile") CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
     "src/lib/Makefile") CONFIG_FILES="$CONFIG_FILES src/lib/Makefile" ;;
     "src/tools/Makefile") CONFIG_FILES="$CONFIG_FILES src/tools/Makefile" ;;
-    "docs/Makefile") CONFIG_FILES="$CONFIG_FILES docs/Makefile" ;;
     "libcares.pc") CONFIG_FILES="$CONFIG_FILES libcares.pc" ;;

   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
@@ -34698,5 +34697,3 @@ done
 ## -------------------------------- ##
 ##  End of distclean amending code  ##
 ## -------------------------------- ##
-
-
diff --git a/deps/c-ares/Makefile.in b/deps/c-ares/Makefile.in
index 35994e4d..f75287f3 100644
--- a/deps/c-ares/Makefile.in
+++ b/deps/c-ares/Makefile.in
@@ -401,7 +401,7 @@ EXTRA_DIST = AUTHORS CHANGES README.cares $(man_MANS) RELEASE-NOTES	\

 CLEANFILES = $(PDFPAGES) $(HTMLPAGES)
 DISTCLEANFILES = include/ares_build.h
-DIST_SUBDIRS = include src test docs
+DIST_SUBDIRS = include src
 SUBDIRS = @BUILD_SUBDIRS@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcares.pc
diff --git a/deps/c-ares/Makefile.am b/deps/c-ares/Makefile.am
index eef3d3d1..c3e37f73 100644
--- a/deps/c-ares/Makefile.am
+++ b/deps/c-ares/Makefile.am
@@ -16,7 +16,7 @@ CLEANFILES = $(PDFPAGES) $(HTMLPAGES)
 
 DISTCLEANFILES = include/ares_build.h
 
-DIST_SUBDIRS = include src test docs
+DIST_SUBDIRS = include src
 
 SUBDIRS = @BUILD_SUBDIRS@
 
diff --git a/deps/c-ares/configure.ac b/deps/c-ares/configure.ac
index 703bde01..06c02c63 100644
--- a/deps/c-ares/configure.ac
+++ b/deps/c-ares/configure.ac
@@ -909,7 +909,7 @@ fi
 AC_MSG_RESULT([$build_tests])
 
 
-BUILD_SUBDIRS="include src docs"
+BUILD_SUBDIRS="include src"
 if test "x$build_tests" = "xyes" ; then
   AC_CONFIG_SUBDIRS([test])
   BUILD_SUBDIRS="${BUILD_SUBDIRS} test"
@@ -922,7 +922,6 @@ AC_CONFIG_FILES([Makefile           \
                  src/Makefile       \
                  src/lib/Makefile   \
                  src/tools/Makefile \
-                 docs/Makefile      \
                  libcares.pc ])
 
 AC_OUTPUT
diff --git a/deps/c-ares/configure b/deps/c-ares/configure
index 50c8266b..96387b37 100755
--- a/deps/c-ares/configure
+++ b/deps/c-ares/configure
@@ -31848,13 +31848,13 @@ fi
 $as_echo "$build_tests" >&6; }
 
 
-BUILD_SUBDIRS="include src docs"
+BUILD_SUBDIRS="include src"
 if test "x$build_tests" = "xyes" ; then
 
 
 subdirs="$subdirs test"
 
-  BUILD_SUBDIRS="${BUILD_SUBDIRS} test"
+  BUILD_SUBDIRS="${BUILD_SUBDIRS}"
 fi
 
 
