diff --git a/deps/libuv/src/win/udp.c b/deps/libuv/src/win/udp.c
index 1c4977af..958dee28 100644
--- a/deps/libuv/src/win/udp.c
+++ b/deps/libuv/src/win/udp.c
@@ -774,8 +774,8 @@ int uv__udp_set_source_membership6(uv_udp_t* handle,
     mreq.gsr_interface = 0;
   }
 
-  STATIC_ASSERT(sizeof(mreq.gsr_group) >= sizeof(*multicast_addr));
-  STATIC_ASSERT(sizeof(mreq.gsr_source) >= sizeof(*source_addr));
+//  STATIC_ASSERT(sizeof(mreq.gsr_group) >= sizeof(*multicast_addr));
+//  STATIC_ASSERT(sizeof(mreq.gsr_source) >= sizeof(*source_addr));
   memcpy(&mreq.gsr_group, multicast_addr, sizeof(*multicast_addr));
   memcpy(&mreq.gsr_source, source_addr, sizeof(*source_addr));
 
@@ -846,7 +846,7 @@ int uv_udp_set_source_membership(uv_udp_t* handle,
                                           src_addr6,
                                           membership);
   }
-  
+
   err = uv_ip4_addr(source_addr, 0, src_addr4);
   if (err)
     return err;
